src/Interpreters/Cache/Metadata.cpp

```diff
@@ -337,7 +337,7 @@ friend struct CacheMetadata;
     {
         {
             std::lock_guard lock(mutex);
-            queue.emplace(file_segment->key(), file_segment->offset(), file_segment);
+            queue.push(DownloadInfo{file_segment->key(), file_segment->offset(), file_segment});
```

Comment:
This is needed to compile with `clang-15`. Unfortunately this is still the latest available version on Manjaro Linux.

Reply:
You can always build the latest clang-17 from master:

```
git clone git@github.com:llvm/llvm-project.git
mkdir llvm-build
cd llvm-build
cmake -GNinja -DCMAKE_BUILD_TYPE:STRING=Release -DLLVM_ENABLE_PROJECTS=all -DLLVM_TARGETS_TO_BUILD=all ../llvm-project/llvm
time ninja
sudo ninja install
```

Linux package managers are irrelevant to modern software development.